File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if nums == []:
            return 0
        maxLength = 1
        for i in nums:
            length = 1
            j = i
            if i-1 not in nums:
                while(j+1 in nums):
                    length+=1
                    j = j+1
            maxLength = max(length, maxLength)
        return maxLength

File: Data Structures & Algorithms/combination-target-sum/submission-1.cpp
class Solution {
public:
    vector<int> path;
    vector<vector<int>> result;
    int sum = 0;
    void dfs(vector<int>& nums, int index, int target){
        if(sum==target){
            result.push_back(path);
        }
        if(sum>target){
            return;
        }

        for(int i=index;i<nums.size();i++){
            path.push_back(nums[i]);
            sum+=nums[i];
            dfs(nums,i,target);
            path.pop_back();
            sum-=nums[i];
        }
    }
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        dfs(nums, 0, target);
        return result;
    }
};
